tools/biotop.py

```diff
@@ -128,7 +128,19 @@ def signal_ignore(signal, frame):
     struct info_t info = {};
     info.major = req->rq_disk->major;
     info.minor = req->rq_disk->first_minor;
+/*
+ * The following deals with a kernel version change (in mainline 4.7, although
+ * it may be backported to earlier kernels) with how block request write flags
+ * are tested. We handle both pre- and post-change versions here. Please avoid
+ * kernel version tests like this as much as possible: they inflate the code,
+ * test, and maintenance burden.
+ */
+#ifdef REQ_WRITE
     info.type = req->cmd_flags & REQ_WRITE;
+#else
+    info.type = (req->cmd_flags >> REQ_OP_SHIFT) & REQ_OP_WRITE;
```

Comment:
Right, thanks, so that's why I did biosnoop that way... It's a simplification (the tool only reports either R or W), and the man page does say that. The biotop man page should match. And there's probably a need for a "biotype" tool that can dump all operation and flag types in human readable format -- or maybe better as a "-v" verbose option to biosnoop.